pkg/uploader/provider/provider.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/vmware-tanzu/velero/internal/credentials"
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+)
+
+const restoreProgressCheckInterval = 10 * time.Second
+const backupProgressCheckInterval = 10 * time.Second
+
+type UploaderProvider interface {
```

Comment:
Rename `UploaderProvider` to `Provider` as the pkg name `uploader/provider` has already indicates it's for `Uploader`

Reply:
Could you add comments to the definitions of interface and functions to explain how to use/implement them?

Reply:
UploaderProvider has changed to Provider
kopiaUploaderProvider has changed to kopiaProvider
resticUploaderProvider has changed to resticProvider

I've added some explanations for some functions as well